### PR TITLE
implement shallow /projects route

### PR DESCRIPTION
**Pull request by ashleygwilliams**

### CODE REVIEW

File: services/api/handlers/projects.js [hunk 1]
```diff
@@ -49,6 +49,25 @@ exports.post = {
 };
 
 exports.get = {
+  oneById: function(request, reply) {
+    request.server.methods.projects.findOneById(
+      request.params.project,
```
Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: @cadecairos does this have to be an array?

Comment (cadecairos) [inline services/api/handlers/projects.js]: yup, it's passing it directly into the pg query function, which expects an array

File: services/api/handlers/projects.js [hunk 2]
```diff
@@ -49,6 +49,25 @@ exports.post = {
 };
 
 exports.get = {
+  oneById: function(request, reply) {
+    request.server.methods.projects.findOneById(
```
Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: for ref: 
`findOneById`: https://github.com/mozilla/api.webmaker.org/blob/develop/services/api/lib/queries.js#L113-L114

File: services/api/handlers/projects.js [hunk 3]
```diff
@@ -49,6 +49,25 @@ exports.post = {
 };
 
 exports.get = {
+  oneShallow: function(request, reply) {
+    request.server.method.projects.findOneShallow(
```
Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: erroring on this line

**Comment by jbuck** [comment]
This code looks good to me, but could you add a test to verify that fetching a project works as expected?

File: test/services/api/handlers/projects.js [hunk 4]
```diff
@@ -446,6 +446,19 @@ experiment('Project Handlers', function() {
...
+  experiment('GET - one project, shallow', function() {
+    test('default', function(done) {
+      var opts = configs.get.findOneShallow.success.default;
+
+      server.inject(opts, function(resp) {
+        expect(resp.statusCode).to.equal(200);
+        expect(resp.result.status).to.equal('success');
+        expect(resp.result.project).to.exist();
+        expect(resp.result.project).to.be.an.object();
+      });
```
Comment (jbuck) [inline test/services/api/handlers/projects.js]: Add a call to `done()` here and it shouldn't timeout

File: services/api/handlers/projects.js [hunk 5]
```diff
@@ -49,6 +49,25 @@ exports.post = {
 };
 
 exports.get = {
+  oneShallow: function(request, reply) {
+    request.server.methods.projects.findOneShallow(
+      [ request.params.project ],
+      function(err, result) {
+        if (err) {
```
Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: @cadecairos @jbuck can't figure out how i'd test this so thinking of removing it

Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: fixed with last commit (force push)

**Comment by jbuck** [comment]
This looks good! The only remaining issue is that the shallow project representation is not the same as the deep project representation. It's missing the `authors` object, which is the user model.

File: test/services/api/handlers/projects.js [hunk 6]
```diff
@@ -447,6 +447,55 @@ experiment('Project Handlers', function() {
...
+  experiment('GET - one project, shallow', function() {
+    test('default', function(done) {
+      var opts = configs.get.findOneShallow.success.default;
+
+      server.inject(opts, function(resp) {
+        expect(resp.statusCode).to.equal(200);
+        expect(resp.result.status).to.equal('success');
+        expect(resp.result.project).to.exist();
+        expect(resp.result.project).to.be.an.object();
+        done();
```
Comment (jbuck) [inline test/services/api/handlers/projects.js]: Can you add one more check here to ensure that the `authors` key exists?

**Comment by jbuck** [comment]
@ashleygwilliams fix up those two little nits and this is good to go :ship:

File: test/services/api/handlers/projects.js [hunk 7]
```diff
@@ -447,6 +447,56 @@ experiment('Project Handlers', function() {
...
+  experiment('GET - one project, shallow', function() {
+    test('default', function(done) {
+      var opts = configs.get.findOneShallow.success.default;
+
+      server.inject(opts, function(resp) {
+        expect(resp.statusCode).to.equal(200);
+        expect(resp.result.status).to.equal('success');
+        expect(resp.result.project).to.exist();
+        expect(resp.result.project).to.be.an.object();
+        expect(resp.result.project.author).to.be.an.object();
```
Comment (ashleygwilliams) [inline test/services/api/handlers/projects.js]: there ya go @jbuck

**Pull request merged**